src/planner/common.go

```diff
@@ -38,3 +41,89 @@ func isUpdateShardKey(exprs sqlparser.UpdateExprs, shardkey string) bool {
 	}
 	return false
 }
+
+// checkField is used to check if the field's Qualifier is illegal or not in where or order by clause.
+func checkField(database, table string, node sqlparser.SQLNode, log *xlog.Log) error {
+	switch node.(type) {
+	case *sqlparser.Delete:
+		// Here we not should do check on limit clause, leave it to the backend mysql.
+		nodePtr := node.(*sqlparser.Delete)
+		if err := checkFieldImpl(database, table, nodePtr.Where, "where clause"); err != nil {
+			return err
+		}
+		if err := checkFieldImpl(database, table, nodePtr.OrderBy, "order clause"); err != nil {
+			return err
+		}
+	default:
+		log.Warning("currently.check.field.only.support.delete.statement.")
+	}
+	return nil
+}
+
+// checkFieldImpl is an implementation of checkField() function.
+func checkFieldImpl(database, table string, node sqlparser.SQLNode, suffix string) error {
+	err := sqlparser.Walk(func(node sqlparser.SQLNode) (kontinue bool, err error) {
+		switch node.(type) {
+		case *sqlparser.ColName:
+			nodePtr := node.(*sqlparser.ColName)
+			colDB := nodePtr.Qualifier.Qualifier.String()
+			colTbl := nodePtr.Qualifier.Name.String()
+			colName := nodePtr.Name.String()
+
+			if colDB != "" && colTbl != "" {
+				// case 1: where/order by db.t.a
+				if colDB != database || colTbl != table {
+					badField := fmt.Sprintf("%s.%s.%s", colDB, colTbl, colName)
+					return false, sqldb.NewSQLError(sqldb.ER_BAD_FIELD_ERROR, badField, suffix)
+				}
+			} else if colDB == "" && colTbl != "" {
+				// case 2: where/order by t.a
+				if colTbl != table {
+					badField := fmt.Sprintf("%s.%s", colTbl, colName)
+					return false, sqldb.NewSQLError(sqldb.ER_BAD_FIELD_ERROR, badField, suffix)
+				}
+			}
+			// case 3: where/order by a, return
```

Comment:
Delete `return`, this is about the case which return from backends, not about the function

Reply:
return had been deleted.